app/src/main/java/com/example/passman/Interactors/VaultViewModel.kt

```diff
@@ -23,22 +49,76 @@ class VaultViewModel(activity: Activity) : ViewModel() {
 
     private val encryptedStorage = VaultKeysStorage(activity)
 
+    // Instantiate the cache
+    val cache = DiskBasedCache(activity.cacheDir, 1024 * 1024) // 1MB cap
+
+    // Set up the network to use HttpURLConnection as the HTTP client.
+    val network = BasicNetwork(HurlStack())
+
+    // Instantiate the RequestQueue with the cache and network. Start the queue.
+    val requestQueue = RequestQueue(cache, network).apply {
+        start()
+    }
+
+    //todo: change to https
+    val baseUrl = "http://192.168.0.12:3000"
+
+
     init {
-        // TODO: api call -> on start fetch tha from api
-        vaults = vaults + defaultVaultData
+        val endpoint = "/vault/get"
+
+        val pubks = encryptedStorage.getAllPublicKeys()
+
+        for(pubk in pubks)
+        {
+            val privk = encryptedStorage.read(pubk)
+
+            val req = makeReq(mapOf(
+                "public_key" to pubk,
+            ),pubk)
+
+            invokeReq(baseUrl+endpoint, req,{ response ->
+                val r = Json.decodeFromString<ApiVaultGetResponse>(response.toString())
+                if(r.status == "success")
+                {
+                    vaults = vaults + VaultData(r.data.name,pubk,
+                        r.data.keys.map { it.name to RSAHelper().decryptWithPrivate(EncodedRSAKeys(pubk,privk),it.value) }.toMutableStateMap())
+                }
+                else {
+                    // TODO: Handle error
+                    Log.d("ERROR MAKING REQ: ", r.status)
+                }
+            } )
+        }
     }
 
     fun createVault(name: String) {
+        val endpoint = "/vault/new"
         val keyPair = KeyPairGenerator().generate()
 
         Log.d("SHARE START PK: ",keyPair.publicKey)
 
         encryptedStorage.write(keyPair.publicKey,keyPair.privateKey)
-        // TODO: api call -> create new vault && vaults = vaults + new vault
-        // but don't touch private key
 
-        // temp should be invoked in response to api call
-        vaults = vaults + listOf(VaultData(name, keyPair.publicKey, mutableMapOf()))
+        val req = makeReq(mapOf(
+            "public_key" to keyPair.publicKey,
+            "name" to name
+        ),keyPair.publicKey)
+
+        invokeReq(baseUrl+endpoint,req,{ response ->
+            val r = Json.decodeFromString<SuccessResponse>(response.toString())
+            if(r.status == "success")
```

Comment:
Ty no, a nie da się tego wszystkiego wydzielić do innej klasy? Bez sensu wszystko w jeden klasie pakować

Reply:
W sensie calej tej logiki komunikacji z serwerem, takie trochę spaghetti :)

Reply:
Mejbi lejter się wydzieli bo to już jest tak opakowane że można ale najpierw przetestuje to